### ISSUE TITLE
MySQL Workbench update doesn't exist

### ISSUE THREAD

**Issue by ffurger** [mangerlahn/Latest#378]
Latest thinks there is a newer version of MySQL Workbench, v. 8.0.36. 
However, as far as I can tell there is only a version 8.0.36.CE, which seems to be the latest version.
Note that trying to check for updates in Workbench invariably produces an error message.

**Comment by chucklessmith** [comment]
Noticing this too. Looks like (one possible | the) culprit line is this comparison of installed vs. remote version numbers?

https://github.com/mangerlahn/Latest/blob/main/Latest/Model/Update.swift#L52

Seems as though 8.0.36.CE (CE = Community Edition, I'm gonna guess) is evaluating as *less than* 8.0.36. 

I could see proposing to truncate trailing non-numerical values when they're at the very end, but I suspect that'll cause things to go wrong in other instances; I see a few examples on my machine of a version number ending with some meaningful alphabetical values.

Maybe check to see if one version ends in a numeral and the other doesn't, and if so, compare just the numerical values; else compare the entire values?

Tempted just to go ahead and test this to see whether this works based on my limited test cases. We'll see when time affords!

**Comment by chucklessmith** [comment]
I do have a build that works around this issue.

I'm not sure what else I've broken in the process -- I'm running the stable release version as well as my alteration, and so far the only discrepancy I've encountered is this MySQL Workbench issue (i.e., it's solved, but I'm still not sure at what cost).

I'll happily share the change if that's of interest, but I'm wary about proposing a pull request, because it's entirely possible I also broke a ton of edge cases...

**Comment by ffurger** [comment]
Thank you, but I’ll pass on this.

**Comment by jifbrodeur** [comment]
Hi, in fact the issue seems to be related to the .CE suffix of the community edition, IMHO latest should drop the .CE suffix when doing the version comparison the SQL Workbench.
Thanks

**Comment by chucklessmith** [comment]
Agreed @jifbrodeur -- For this particular case, yeah, dropping the version-final alphabetical characters works for the new-version-exists logic, but there are several other edge cases each being separately accounted for, and the code suggests there may need to be a whole rethink on the version comparisons into something that works more universally. That would, of course, be best as opposed to writing update logic for every nonstandard or unusual version numbering system out there.

As I mentioned earlier, though, I do have a build that solves the MySQL Workbench issue specifically.

**Comment by jifbrodeur** [comment]
@chucklessmith Your point of view is very valid, writing a switch or if for every exception might not be a good way to implement a maintainable solution. Maybe having two version of the software the CE and the non CE version could be a better way to resolve the issue, that is If Oracle always append a .CE at the end of the Community Edition version of MySQL Workbench?

**Comment by Gana-Dube** [comment]
So the main suggestion is to drop the .CE suffix from the MySQL Workbench version number when performing version comparisons to ensure the update logic works correctly right?

`^(\d+\.\d+\.\d+)(?:\.CE)?$`

![image](https://github.com/user-attachments/assets/7cdfacfe-10f2-4003-8fea-66fdac7d8cdc)
![image](https://github.com/user-attachments/assets/7bf03a54-97c1-4cb4-a827-11409c43d32e)

**Comment by jifbrodeur** [comment]
That solution would work, but it wouldn’t solve the larger problem of having to write an if statement for every odd software release that a developer creates. My suggestion of having two versions one CE and one regular seems like an easier alternative?

**Comment by chucklessmith** [comment]
Just noticed they've apparently removed the .CE suffix from Community Edition version numbers. Assuming that's policy (can't find evidence of this, but admittedly haven't looked too hard, beyond the brief changelogs on Oracle's site) then this specific issue can be closed, general nonstandard version number issues notwithstanding.

**Comment by chucklessmith** [comment]
Please disregard my earlier comment regarding the version numbers. The issue persists after the update.

**Comment by woj-tek** [comment]
I just found about Latest and unfortunately run into the issue with weird versions/suffixes:

![Image](https://github.com/user-attachments/assets/18b42f84-faee-448f-a66d-d8ce1769b0cb)

I do use Homebrew to install versions and it seems to completely do away without suffixes (keeps track of what was installed without actually relying on that version app reports?):

https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/m/mysqlworkbench.rb

**Comment by mangerlahn** [comment]
Sorry for being quiet on this. Version comparison is a big mess, I already started to implement individual fixes for certain apps. Yes, implementing the suffix thing might affect other apps, but we can only fix whats reproted (at least for now 😅). So, one more exception does not hurt, I also use regex here: https://github.com/mangerlahn/Latest/blob/main/Latest/Model/Version/VersionParser.swift

With tests here:
https://github.com/mangerlahn/Latest/blob/main/Tests/VersionParserTest.swift

Feel free to create a PR for this.

**Comment by jifbrodeur** [comment]
Hi, here are a few exceptions that are listed on my list of available app:
Visual Studio Code Insider my version : 1.100.0-insider new: 1.100.0
Microsoft Edge my v: 136.0.3240.50 (136.0.3240.25050150) new: 136.0.3240.50 (138cbf57)
classic MySQL Workench wiht the suffix CE
Sonos my v: 16.3 New 84.1 but they are the same version
And what do we do with Snagit 2024.4.3 that still is the only perpetual licence but shows that 2025.1.0 is the new version.
Thanks for all the hard work and as soon as a new version of Latest is available I'll gladly test it.
Jean-François
